PEA/MergeIfElseParanoid.java

```diff
@@ -0,0 +1,26 @@
+class MergeIfElseParanoid {
+    static class Node {
+       public Node _next;
+       public Node() {}
+    }
+
+    static Object _global;
+    static Object _global2;
+
+    public static void test(boolean cond) {
+        Object foo = new Object();
+        if (cond) {
+            _global = foo;
+        }
+        // When `cond == true`, we materialize foo a second time. We cannot
+        // eliminate the original allocation because we need for when
+        // `cond == false`.
+        _global2 = foo;
```

Comment:
if we have passive materialization, C2 PEA materializes the object foo in its predecessor.  the original object will be dead. we don't have this duplication.

Reply:
+1. I added more details to the comment.